File: QTi/frontend/src/types/bot.ts
export interface BotStats {
  totalBots: number;
  activeBots: number;
  totalProfit: number;
  totalTrades: number;
  winRate: number;
  averageProfit: number;
  maxDrawdown: number;
  sharpeRatio: number;
}

export interface BotStatus {
  id: string;
  name: string;
  exchange: string;
  symbol: string;
  status: 'running' | 'stopped' | 'error';
  lastUpdate: string;
  lastPrice?: number;
  lastError?: string;
  profit: number;
  trades: number;
  winRate: number;
  config: BotConfig;
}

export interface BotConfig {
  strategy: string;
  timeframe: string;
  leverage?: number;
  positionSize?: number;
  stopLoss?: number;
  takeProfit?: number;
  trailingStop?: boolean;
  trailingStopDistance?: number;
  maxPositions?: number;
  maxDrawdown?: number;
  riskPerTrade?: number;
}

export interface BotTrade {
  id: string;
  botId: string;
  symbol: string;
  side: 'long' | 'short';
  entryPrice: number;
  exitPrice?: number;
  quantity: number;
  profit?: number;
  status: 'open' | 'closed';
  entryTime: string;
  exitTime?: string;
  stopLoss?: number;
  takeProfit?: number;
}

export interface BotPerformance {
  equityCurve: {
    timestamp: string;
    equity: number;
  }[];
  trades: BotTrade[];
  stats: {
    totalTrades: number;
    winRate: number;
    averageProfit: number;
    maxDrawdown: number;
    sharpeRatio: number;
    profitFactor: number;
    expectancy: number;
  };
} 

File: QTi/backend/app/models/bot.py
from pydantic import BaseModel, Field
from typing import Optional, List, Literal
from datetime import datetime

class Position(BaseModel):
    side: Literal["long", "short", "none"]
    size: float
    entry_price: float
    current_price: float

class BotStatus(BaseModel):
    id: str
    name: str
    symbol: str
    status: Literal["running", "stopped", "error"]
    pnl: float
    position: Position
    last_update: datetime

class Trade(BaseModel):
    id: str
    bot_id: str
    symbol: str
    side: Literal["buy", "sell"]
    price: float
    size: float
    timestamp: datetime
    pnl: float

class BotConfig(BaseModel):
    name: str
    symbol: str
    timeframe: str
    strategy: str
    parameters: dict
    risk_management: dict
    enabled: bool = True

class Bot(BaseModel):
    id: str
    config: BotConfig
    status: BotStatus
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

class BotCreate(BaseModel):
    config: BotConfig

class BotUpdate(BaseModel):
    config: Optional[BotConfig] = None
    enabled: Optional[bool] = None 